oval_graph/xml_parser.py

```diff
@@ -109,20 +108,16 @@ def _get_oval_definitions(self):
             'oval-definitions:definitions', ns)
 
     def _get_definition_of_rule(self, rule_id):
-        for definition in self.scan_definitions['definitions']:
-            if self.get_def_id_by_rule_id(rule_id) == definition['id']:
-                return dict(rule_id=rule_id, definition=definition)
-
-    def get_def_id_by_rule_id(self, rule_id):
-        for rule in self.notselected_rules:
-            if rule['id_rule'] == rule_id:
-                raise ValueError(
-                    'err- rule "{}" was not selected, so there are no results.'
-                    .format(rule_id))
-        for rule in self.used_rules:
-            if rule['id_rule'] == rule_id:
-                return rule['id_def']
-        raise ValueError('err- 404 rule not found!')
+        if rule_id in self.used_rules:
+            return dict(rule_id=rule_id,
+                        definition_id=self.used_rules[rule_id]['id_def'],
+                        definition=self.scan_definitions[self.used_rules[rule_id]['id_def']])
+        elif rule_id in self.notselected_rules:
+            raise ValueError(
+                'err- rule "{}" was not selected, so there are no results.'
```

Comment:
Let's make this more user-friendly and representative (in another PR) - current error messages are quite geeky, and the client should not show the traceback, but it should only print the message.

Reply:
I create issue for this.